src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +153,1304 @@ static DamageTable D_80B65044 = {
+    Math_StepToF(&this->actor.speedXZ, 10.0f, 1.0f);
+    Math_SmoothStepToF(&this->actor.world.pos.y, player->actor.world.pos.y, 0.3f, 7.5f, 1.0f);
+    EnBigpo_UpdateSpin(this);
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    // because acFlags AC_HARD and COLTYPE_METAL, if we hit it means we contacted as attack
+    if (((this->collider.base.atFlags & AT_HIT)) ||
+        ((((yawDiff < 0) ? -yawDiff : yawDiff) >= 0x4001) && (this->actor.xzDistToPlayer > 50.0f))) {
+        // hit the player OR the poe has missed and flew past player
+        EnBigpo_SetupSpinDown(this);
+    }
+}
+
+// after missing, spin down before idle flying
+void EnBigpo_SetupSpinDown(EnBigpo* this) {
+    this->collider.base.atFlags &= ~AT_ON;
+    this->actionFunc = EnBigpo_SpinningDown;
+}
+
+void EnBigpo_SpinningDown(EnBigpo* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothStepToF(&this->actor.world.pos.y, player->actor.world.pos.y + 100.0f, 0.3f, 5.0f, 1.0f);
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f);
+    if (Math_ScaledStepToS(&this->rotVelocity, 0, 0x200) != 0) {
+        // spin down complete, re-allow hittable
+        this->collider.base.colType = COLTYPE_HIT3;
+        this->collider.base.acFlags &= ~AC_HARD;
+        this->collider.info.bumper.dmgFlags |= 0x8000;
+        EnBigpo_SetupIdleFlying(this);
+    }
+    EnBigpo_UpdateSpin(this);
+}
+
+// hit by player, stunned
+void EnBigpo_HitStun(EnBigpo* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06000454, -6.0f);
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 0x10);
+    this->collider.base.acFlags &= ~AC_ON;
+    func_800BE504(&this->actor, &this->collider);
+    this->actionFunc = EnBigpo_CheckHealth;
+    this->actor.speedXZ = 5.0f;
+}
+
+// check if just damaged or dead
+void EnBigpo_CheckHealth(EnBigpo* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.5f);
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnBigpo_SetupDeath(this);
+        } else {
+            EnBigpo_SetupWarpOut(this);
+        }
+    }
+}
+
+void EnBigpo_SetupDeath(EnBigpo* this) {
+    this->idleTimer = 0;
+    this->actor.speedXZ = 0.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.hintId = 0xFF; // clear?
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    this->actionFunc = EnBigpo_BurnAwayDeath;
+}
+
+// from [red damaged poe] to [burning up poe] to [shinking into the lantern]
+void EnBigpo_BurnAwayDeath(EnBigpo* this, GlobalContext* globalCtx) {
+    Vec3f tempVec;
+    f32 unkTemp2; // dont really know what these unktemps are doing
+    s16 cam;
+    s16 unkTemp;
+    s16 modifiedTimer;
+
+    this->idleTimer += 1;
+    if (this->idleTimer < 8) {
+        cam = func_800DFCDC(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x4800;
```

Comment:
```suggestion
        cam = func_800DFCDC(ACTIVE_CAM) + 0x4800;
```
Just some other useful camera macros